crates/bevy_reflect/src/type_registry.rs

```diff
@@ -406,6 +406,15 @@ impl TypeRegistry {
     pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut TypeRegistration> {
         self.registrations.values_mut()
     }
+
+    /// Returns an iterator over the [`TypeRegistration`]s with their corresponding [`TypeData`]
+    /// of the registered types.
+    pub fn iter_with_data<T: TypeData>(&self) -> impl Iterator<Item = (&TypeRegistration, &T)> {
```

Comment:
Mild preference for `TD` as the generic here.

Reply:
I was trying to be consistent with other methods (like `get_type_data`).